Automattic/jetpack
# Connection: Fix token validation logic

Fixes the token validation logic.

#### Changes proposed in this Pull Request:

* Tokens: Updates the `validate` method to return `false` in case any of the tokens to be validated is missing before performing the remote request.
* Manager: Updates the `restore` method to stricter check the `validate` return type
* Adds unit tests

#### Jetpack product discussion


1191179647901802-as-1200123916753267/f

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

Various broken token / reconnection flows:

- Using the Jetpack Debug tools (Broken Token utilities) try clearing / invalidating the user tokens and confirm the reconnection flow works (you should be asked to connect your WPCOM account)
- Repeat the above as a secondary not-connected user (you should be asked to connect your WPCOM account)
- Invalidate the blog token and confirm that the reconnection flow works (you should NOT be asked to connect your WPCOM account)
- Disconnect your site and connect in user-less mode
- Invalidate the blog token and confirm that the reconnection flow works (you should NOT be asked to connect your WPCOM account)
